ed448-goldilocks/src/edwards/extended.rs

```diff
@@ -950,14 +654,14 @@ impl Mul<&EdwardsScalar> for &EdwardsPoint {
 
     /// Scalar multiplication: compute `scalar * self`.
     fn mul(self, scalar: &EdwardsScalar) -> EdwardsPoint {
-        self.scalar_mul(scalar)
+        self.to_affine().scalar_mul(scalar)
     }
```

Comment:
This in particular seems problematic. Using the map from #1316 you can convert directly from the untwisted to twisted form without an intermediate affine conversion.

So scalar multiplication, and all of the functionality associated with it which has been removed in this PR, should remain here on `EdwardsPoint`, which IMO should be the main user-facing API the library presents (ala `curve25519-dalek`).

I think it's okay to have scalar multiplication on `AffinePoint`, but it should work by homogenizing into `EdwardsPoint` first, then performing scalar multiplication with that.

Reply:
Yes indeed, now that we have the direct mapping I will rebase this PR so it makes more sense.

Reply:
Done.